--- ./Lib/V2Protocol.c.orig	2012-05-07 06:57:03.584546979 +0200
+++ ./Lib/V2Protocol.c	2012-05-07 07:19:54.005270428 +0200
@@ -144,7 +144,7 @@
 	wdt_disable();
 
 	Endpoint_WaitUntilReady();
-	Endpoint_SelectEndpoint(AVRISP_DATA_OUT_EPNUM);
+	Endpoint_SelectEndpoint(AVRISP_DATA_OUT_EPNUM__COMMON);
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_OUT);
 }
 
@@ -163,7 +163,13 @@
 	}
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);
+	} else {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);
+	}
+
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	Endpoint_Write_8(V2Command);
@@ -175,7 +181,14 @@
 static void V2Protocol_SignOn(void)
 {
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+
+	if ( use_libusb == true ) {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);
+
+	} else {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);
+	}
+
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	Endpoint_Write_8(CMD_SIGN_ON);
@@ -191,7 +204,14 @@
 static void V2Protocol_ResetProtection(void)
 {
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+
+	if ( use_libusb == true ) {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);
+
+	} else {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);
+	}
+
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	Endpoint_Write_8(CMD_RESET_PROTECTION);
@@ -214,7 +234,13 @@
 	  ParamValue = Endpoint_Read_8();
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+
+	if ( use_libusb == true ) {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);
+	} else {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);
+	}
+
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	Endpoint_Write_8(V2Command);
@@ -248,7 +274,13 @@
 	Endpoint_Read_Stream_BE(&CurrentAddress, sizeof(CurrentAddress), NULL);
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+
+	if ( use_libusb == true ) {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);
+	} else {
+		Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);
+	}
+
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	if (CurrentAddress & (1UL << 31))
