--- ./Lib/XPROG/XPROGProtocol.c.orig	2012-05-07 06:57:20.375982879 +0200
+++ ./Lib/XPROG/XPROGProtocol.c	2012-05-07 07:21:29.486017966 +0200
@@ -65,7 +65,13 @@
 	Endpoint_Read_Stream_LE(&SetMode_XPROG_Params, sizeof(SetMode_XPROG_Params), NULL);
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}
+
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	XPROG_SelectedProtocol = SetMode_XPROG_Params.Protocol;
@@ -112,7 +118,13 @@
 static void XPROGProtocol_EnterXPROGMode(void)
 {
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}	
+	
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	bool NVMBusEnabled = false;
@@ -134,7 +146,13 @@
 static void XPROGProtocol_LeaveXPROGMode(void)
 {
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}
+	
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	if (XPROG_SelectedProtocol == XPRG_PROTOCOL_PDI)
@@ -169,7 +187,13 @@
 	Erase_XPROG_Params.Address = SwapEndian_32(Erase_XPROG_Params.Address);
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}
+
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	uint8_t EraseCommand;
@@ -260,7 +284,13 @@
 	}
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}
+	
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	if (XPROG_SelectedProtocol == XPRG_PROTOCOL_PDI)
@@ -342,7 +372,13 @@
 	ReadMemory_XPROG_Params.Length  = SwapEndian_16(ReadMemory_XPROG_Params.Length);
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}
+	
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	uint8_t ReadBuffer[256];
@@ -385,7 +421,13 @@
 	Endpoint_Read_Stream_LE(&ReadCRC_XPROG_Params, sizeof(ReadCRC_XPROG_Params), NULL);
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}
+	
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	uint32_t MemoryCRC;
@@ -467,7 +509,13 @@
 	}
 
 	Endpoint_ClearOUT();
-	Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM);
+	
+	if ( use_libusb == true ) {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__LIBUSB);	  
+	} else {
+	    Endpoint_SelectEndpoint(AVRISP_DATA_IN_EPNUM__DEFAULT);	  
+	}
+	
 	Endpoint_SetEndpointDirection(ENDPOINT_DIR_IN);
 
 	Endpoint_Write_8(CMD_XPROG);
